RSSReader/Services/FileManager.m

```diff
@@ -0,0 +1,61 @@
+//
+//  FileManager.m
+//  RSSReader
+//
+//  Created by Uladzislau Stankevich on 25.02.21.
+//
+
+#import "FileManager.h"
+
+
+@interface FileManager ()
+
+@property (nonatomic, retain) NSString *path;
+
+@end
+
+@implementation FileManager
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        _path = [[NSString alloc] initWithFormat:@"%@/History", [NSHomeDirectory() stringByAppendingPathComponent:@"Documents"]];
+    }
+    return self;
+}
+
+- (void)writeToFile:(SearchFeedItem *)item {
+    
+    if (![self isRepeat:item]) {
+        NSMutableArray *tempArray = [NSMutableArray arrayWithArray:[self readData]];
+        [tempArray addObject:item];
+        
+        NSData *data = [NSKeyedArchiver archivedDataWithRootObject:tempArray requiringSecureCoding:YES error:nil];
+        [data writeToFile:self.path atomically:NO];
+    }
+}
+
+
+- (NSArray<SearchFeedItem *>*)readData {
+    NSData *data = [[NSMutableData alloc]initWithContentsOfFile:self.path];
+    NSArray *entries = [NSKeyedUnarchiver unarchivedArrayOfObjectsOfClass:[SearchFeedItem class] fromData:[data autorelease] error:nil];
```

Comment:
`unarchivedArrayOfObjectsOfClass:fromData:error:` is only available on iOS 14.0 or newer

Reply:
fixed